java/sage/io/LocalSageFile.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2015 The SageTV Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sage.io;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+
+public class LocalSageFile extends RandomAccessFile implements SageFileSource
```

Comment:
I didn't extend it at first, but then I did in what could be a misguided attempt to have one less method between the top class and the actual data source. I can see how someone might accidentally use that incorrectly. I'll make it less exposed.